lib/widgets/login.dart

```diff
@@ -71,13 +183,28 @@ class _AddAccountPageState extends State<AddAccountPage> {
             constraints: const BoxConstraints(maxWidth: 400),
             child: TextField(
               controller: _controller,
-              onSubmitted: (value) => _onSubmitted(context, value),
+              onSubmitted: (value) => _onSubmitted(context),
               keyboardType: TextInputType.url,
+              autocorrect: false,
+              textInputAction: TextInputAction.go,
+              onEditingComplete: () {
+                // Repeat default implementation by clearing IME compose session…
+                _controller.clearComposing();
+                // …but leave out unfocusing the input in case more editing is needed.
+              },
               decoration: InputDecoration(
                 labelText: 'Your Zulip server URL',
-                suffixIcon: InkWell(
-                  onTap: () => _onSubmitted(context, _controller.text),
-                  child: const Icon(Icons.arrow_forward))))))));
+                hintText: 'your-org.zulipchat.com',
+                errorText: error == null || error.shouldDeferFeedback()
+                  ? null
+                  : error.message(),
+                suffixIcon: IconButton(
+                  onPressed: !_inProgress && (error == null || error.shouldDeferFeedback())
+                    ? () => _onSubmitted(context)
+                    : null, // disabled; errorText will be present
```

Comment:
This comment is true when first added, but no longer true after the commit introducing `_inProgress`.

I think best to leave the comment out, and if the logic feels unclear then try to clarify the code itself.

One thing that'd probably be helpful is to make a local `errorText`, so that you can write `errorText: errorText` above. Then the condition here would be `!_inProgress && errorText == null`, which I think explains just as much as the comment does (when the comment is accurate).